### DISCUSSION TITLE
Fresh instalation in a share hosting

### DISCUSSION THREAD

**Discussion by kacelnik** [FreshRSS/FreshRSS#6879] [Q&A]
Hi everyone,
I'm new to FreshRSS. I just installed a copy on a share hosting. Everything smooth an easy so far. 
I was wondering how to secure the installation given that all files are online. I read the suggestion of exposing only "./p/", so I tried to move everything except "p" to a directory call "/secure/" with the idea of password protecting that directory. But I can't find a way to make the necessary modifications (on constants.php?) so that works.

Are there any in detail instructions on how to setup on a shared hosting?
Or, would it be enough security to password protect the "data" directory and leave the rest as it comes by default?

Thank you
a

**Comment by Frenzie** [comment]
On shared hosting you can typically do that kind of thing with .htaccess in the root, assuming it's served by Apache. Note that the files aren't exposed exactly since every folder comes with one already https://github.com/FreshRSS/FreshRSS/blob/b30a8fa652018025f8ccc6b361823486d04cf390/app/.htaccess

The difference is that if you tell the root to serve `/i/p` all of the underlying logic is more or less completely hidden.

**Comment by kacelnik** [comment]
Thank you for the super quick answer!
So if I understand correctly I can leave everything as it is right out of the box, right?

**Reply by Alkarex** [reply]
Assuming your shared hosting uses Apache, yes, it should be fine out of the box. 
You should double check though by trying to access some files: nothing outside `/p/` should work.

**Comment by kacelnik** [comment]
Yes, I'm using an Apache server.
This folders seems to be protected with .htaccess and I content can't be reached via browser:
- app
- cli
- data
- lib

But this others doesn't have .htaccess and it's content can be reached with a browser:
- .devcontainer
- Docker
- docs
- extensions
- tests

Other than that all files on the root can be reached as well: .dockerignore, CHANGELOG.md, composer.json, several readmes, etc.

**Reply by Alkarex** [reply]
@kacelnik Would you be able to test this patch? https://github.com/FreshRSS/FreshRSS/pull/6881

**Reply by Alkarex** [reply]
Please try our rolling release branch `edge`. Feedback welcome

**Reply by kacelnik** [reply]
I'm happy to test edge and give feedback. Can I change the "Release channel" somewhere and run an update or do I need to reinstall everything?

**Reply by Alkarex** [reply]
@kacelnik It depends on the way you deployed / installed FreshRSS? Did you take the ZIP archive and uploaded to your shared server via SFTP, or do you have an SSH access allowing a git install?

**Reply by kacelnik** [reply]
I downloaded the zip, unzipit and then uploaded everything via SFTP. No SSH I'm afraid.

**Reply by Alkarex** [reply]
Two options then (after the usual warning of first taking a backup):

1. Download a [copy of our rolling release](https://github.com/FreshRSS/FreshRSS/archive/refs/heads/edge.zip) and upload the same way than what you did last time, just overriding the existing files.
2. Or if you use Web updates (requiring that the Web server has sufficient write access), you can edit your `constants.php` to add a `-dev` suffix to your existing version number in:
https://github.com/FreshRSS/FreshRSS/blob/4461784b8a519fe5f77e374be8e79818422933a0/constants.php#L7
to become `1.24.3-dev`, on the model of:
https://github.com/FreshRSS/FreshRSS/blob/91624037c7d73eb545478aab2f8abc55fc224453/constants.php#L7
and then proceed with the update from the Web interface

**Comment by kacelnik** [comment]
I think the permission issue is solved.

FYI I tried the update changing the constant.php and at the web update I got a "An update of FreshRSS is available: Version dev. " message. 
I executed the update and got a "Update complete!" message, but every time I open the update page it kept offering the update with the same "Update complete!" result and files didn't seem to update.
So I went to the first options: I upload everything from scratch and then restored the data folder from the previous installation.  
Now my data is back and all directories have an .htaccess. Also when I try to access files on any directories with the browser I get an 403 Forbidden result. I still can access files on the root (some .txt, .json, etc).

Can I safely remove the '-dev' from the contant.php file so
const FRESHRSS_VERSION = '1.25.0-dev';
becomes
const FRESHRSS_VERSION = '1.25.0';
to get back to the stable release?

**Reply by Alkarex** [reply]
You can just stay on the rolling release and decide at the next slow release (1.25.0) whether to continue on the rolling release or switch back to the slow releases.
Otherwise, you can lie and indicate `1.24.3` again.

**Reply by kacelnik** [reply]
Thank you for all your help!
:)

**Reply by Alkarex** [reply]
Reworked in https://github.com/FreshRSS/FreshRSS/pull/7008
